src/allmydata/test/test_cli_magic_folder.py

```diff
@@ -0,0 +1,199 @@
+import os.path
+import re
+import json
+
+from twisted.trial import unittest
+from twisted.internet import defer
+
+from allmydata.util import fileutil
+from allmydata.scripts.common import get_aliases
+from allmydata.test.no_network import GridTestMixin
+from .test_cli import CLITestMixin
+from allmydata.scripts import magic_folder_cli
+from allmydata.util.fileutil import abspath_expanduser_unicode
+from allmydata.frontends.magic_folder import MagicFolder
+
+
+class MagicFolderCLITestMixin(CLITestMixin, GridTestMixin):
+
+    def create_magic_folder(self, client_num):
+        d = self.do_cli("magic-folder", "create", "magic", client_num=client_num)
+        def _done((rc,stdout,stderr)):
+            self.failUnless(rc == 0)
```

Comment:
In general ``failUnless`` should be used very sparingly if at all, because it doesn't print much useful information.